### DISCUSSION TITLE
VENTOY in Disk 0

### DISCUSSION THREAD

**Discussion by nkpro2000sr** [ventoy/Ventoy#788] [Show and tell]
![Disk 0: \[ VENTOY ext2 16GiB \] \[ VTOYEFI fat16 32MiB \] \[ unallocated 68 MiB \] \[ Windows ntfs 100GiB\] \[ ESP fat32 100MiB \] \[ Linux 50GiB \] \[ Storage ntfs 300GiB \]](https://github.com/nkpro2000sr/man-manjaro/raw/master/_static/img/VTinD0.png)

> 2nd partition must be 32MiB to pass a [test](https://github.com/ventoy/Ventoy/blob/b63ce2a3df409eb0dd0bed5b977a75605997fd3c/EDK2/edk2_mod/edk2-edk2-stable201911/MdeModulePkg/Application/Ventoy/Ventoy.c#L538).
> iso files should be in 1st partition (as usual) to be detected by ventoy.

No need to write menu-entry in grub.cfg for ISOs separately anymore. just
```
menuentry "VENTOY" --class ventoy --class bootloader {
        insmod part_gpt
        insmod ext2
        insmod chain
        set root='hd0,gpt2'
        chainloader /efi/boot/bootx64.efi
}
```
and placing ISOs in 1st partition is enough \o/.

**Comment by eL-Squaz** [comment]
HI, I have been looking into this for a bit now trying to figure out just how to exactly setup these files and where exactly to put them.. does this just go in the ventoy_grub.cf , just that block itself and that's it? 

and it just goes in    '/ventoy'           ? like...

/ventoy/ventoy_grub.cfg

and it doesn't matter if the ISO images are in the /ventoy  directory r not?

**Comment by nkpro2000sr** [comment]
@eL-Squaz It is same as first two partitions in USB drive (flashed with VENTOY).
It supports all features of VENTOY, you can place [/ventoy/ventoy.json](https://www.ventoy.net/en/plugin.html) in this first partition.

Simplest way:
1. Create first partition with filesystem exFAT/FAT32/NTFS/UDF/XFS/Ext2/Ext3/Ext4.
2. Create second partition with size 32MiB and FAT filesystem.
3. Copy all files from "second partition in already VENTOY flashed USB drive" to "second partition in disk".

> Please refer https://www.ventoy.net/en/doc_disk_layout.html
